### PR TITLE
Implemented HITS algorithm (hubs and authorities)

### PR DESCRIPTION
**Pull request by nrs1729**

### CODE REVIEW

File: core/src/main/scala/com/raphtory/algorithms/generic/HITS.scala [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+                val outDegree = vertex.outDegree
+                val inDegree  = vertex.inDegree
+
+                if (outDegree > 0.0)
+                  vertex.messageOutNeighbours(HubMsg(newHub))
+                if (inDegree > 0.0)
+                  vertex.messageInNeighbours(AuthMsg(newAuth))
+
+                if (Math.abs(currHub - newHub) < 0.00001 & Math.abs(currAuth - newAuth) < 0.00001)
```
Comment (Alnaimi-) [inline core/src/main/scala/com/raphtory/algorithms/generic/HITS.scala:111]: Can you make the `0.00001` as a default parameter please.

File: core/src/main/scala/com/raphtory/algorithms/generic/HITS.scala [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+        var newAuth = currAuth / authMax
+        var newHub  = currHub / hubMax
+
+        if (truncate == true) {
+          // round to 8 dp
+          val mult: Double = scala.math.pow(10, 8)
+          newAuth = (math floor newAuth * mult) / mult
+          newHub = (math floor newHub * mult) / mult
+        }
```
[the remark this answers is not in the document]
Comment (ljeub-pometry) [reply core/src/main/scala/com/raphtory/algorithms/generic/HITS.scala:152]: See #565

**Review by ljeub-pometry — CHANGES_REQUESTED** [review]
Algorithm looks good. I've opened a pull request for doing floating-point comparison tests (#565). Once that is in, you can remove the truncate option.
The documentation needs some minor fixes and this should be good to merge.

**Review by ljeub-pometry — APPROVED** [review]
Looks good, ready to merge

**Pull request merged**